### DISCUSSION TITLE
Pagination not working when retrieving taxonomy terms inside collection loop

### DISCUSSION THREAD

**Discussion by m-berger-nf** [statamic/cms#6147] [Q&A] [answered]
Hi,
I have a collection for blog posts (blog_posts). Each of the blog posts can have several tags (blog_post_tags) assigned to it.
When looping over the collection and displaying a list of all posts I also show the assigned tags for each of the posts. This is working fine without pagination. However if I am trying to do the same with pagination, pagination isn't working because the paginate variable is empty.
It seems like the problem is with the loop that goes over the assigned tags. If I remove the code for the assigned tags everything is working fine.
This is the not-working code:
```

{{ collection:blog_posts paginate="2" as="posts" }}
 
    {{ if no_results }}
        <p>No blog posts found.</p>
    {{ /if }}

    {{ posts }}
        <div>

            <div>{{ if blog_post_categories }}{{ blog_post_categories:title | upper }}{{ /if }}</div>

            <div>{{ title }}</div>

            {{ if blog_post_tags }}
                <div>
                    {{ blog_post_tags }}
                        <a href="{{ url }}">{{ title | upper }}</a>
                    {{ /blog_post_tags }}
                </div>
            {{ /if }}

            <div>
                {{ summary }}
            </div>

            <div>
                <a href="{{ url }}" class="button-white inline-block px-6 py-4">Read More</a>
            </div>

        </div>
    {{ /posts }}

    {{ paginate }}
        <a href="{{ prev_page }}">Previous</a>
        <a href="{{ next_page }}">Next</a>
    {{ /paginate }}

{{ /collection:blog_posts }}
```
When I remove this part
```
            {{ if blog_post_tags }}
                <div>
                    {{ blog_post_tags }}
                        <a href="{{ url }}">{{ title | upper }}</a>
                    {{ /blog_post_tags }}
                </div>
            {{ /if }}
```
it works.
What am I doing wrong and how would I implement this correctly?

**Comment by JohnathonKoster** [comment, marked as answer]
What you have there should "just work". This is a bug, and will be resolved with this PR: https://github.com/statamic/cms/pull/6153

**Comment by m-berger-nf** [comment]
Thanks for the fast reply @JohnathonKoster and the quick fix. Just updated to v3.3.14 and it's working fine now 👍
